### PR TITLE
Add suggestions (as hints) about missing example.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

**Comment by isoos** [comment]
It doesn't handle the 'too small / too large' example edge cases, but I'd wait for a real-world example before implementing that. Closes #208.

File: lib/src/maintenance.dart [hunk 1]
```diff
@@ -206,6 +223,22 @@ Future<Maintenance> detectMaintenance(
         'Readme should inform others about your project, what it does, and how they can use it.',
         penalty: new Penalty(fraction: 500)));
   }
+  if (!exampleExists) {
+    final exampleDirExists = files.any((file) => file.startsWith('example/'));
+    if (exampleDirExists) {
+      maintenanceSuggestions.add(new Suggestion.hint(
+          'Maintain an example.',
+          'None of the files in your `example/` directory matches a known example patterns. '
```
Comment (kevmoo) [inline lib/src/maintenance.dart]: We could print out a full list of the supported example names, right?

Comment (isoos) [reply lib/src/maintenance.dart]: I wouldn't duplicate the list with the `lib/*` versions, and the `$package_example.dart` seemed a bit too long here. The others are listed here.

**Comment by kevmoo** [comment]
SGTM

**Pull request merged**